sdk/cosmos/src/requests/replace_collection_builder.rs

```diff
@@ -32,28 +24,24 @@ impl<'a, 'b> ReplaceCollectionBuilder<'a, 'b, No, No> {
             user_agent: None,
             activity_id: None,
             consistency_level: None,
-            p_partition_key: PhantomData,
-            p_indexing_policy: PhantomData,
         }
     }
 }
 
-impl<'a, 'b, PartitionKeysSet, IndexingPolicySet>
-    ReplaceCollectionBuilder<'a, 'b, PartitionKeysSet, IndexingPolicySet>
-where
-    PartitionKeysSet: ToAssign,
-    IndexingPolicySet: ToAssign,
-{
+impl<'a, 'b> ReplaceCollectionBuilder<'a, 'b> {
     setters! {
         user_agent: &'b str => Some(UserAgent::new(user_agent)),
         activity_id: &'b str => Some(ActivityId::new(activity_id)),
         consistency_level: ConsistencyLevel => Some(consistency_level),
+        indexing_policy: &'a IndexingPolicy => Some(indexing_policy),
     }
 }
 
-// methods callable only when every mandatory field has been filled
-impl<'a, 'b> ReplaceCollectionBuilder<'a, 'b, Yes, Yes> {
-    pub async fn execute(&self) -> Result<CreateCollectionResponse, CosmosError> {
+impl<'a, 'b> ReplaceCollectionBuilder<'a, 'b> {
+    pub async fn execute<P: Into<PartitionKey>>(
+        &self,
+        partition_key: P,
```

Comment:
@thomastaylor312 actually I think we want to set this on the call to `replace_collection` no?